packages/nimble-components/src/table/types.ts

```diff
@@ -24,3 +24,13 @@ export type TableFieldValue =
 export interface TableRecord {
     [key: TableFieldName]: TableFieldValue;
 }
+
+export interface TableValidity {
+    readonly duplicateRowId: boolean;
+    readonly missingRowId: boolean;
+    readonly invalidRowId: boolean;
+}
+
+export interface TableRowState<TData extends TableRecord = TableRecord> {
+    data: TData;
```

Comment:
I had `id` as part of this interface, but then I decided to remove it since it wasn't needed yet. Ultimately `id` will be in this interface though. I can add it back now if you think that would be better.

Reply:
Your call. I just noticed that we were already returning `TableRowState` instances with the `id` set, so it seemed reasonable for it to be on the interface.

Reply:
I had just noticed that too. For right or for wrong, I went ahead and removed setting `id` when creating the `TableRowState`.